tests/lib/api-request-test.js

```diff
@@ -521,6 +522,64 @@ describe('APIRequest', function() {
 			});
 		});
 
+		it('should retry when a 5xx error occurs and then restore with sucessful response', function(done) {
+			var responseInfo1 = {
+				statusCode: 504,
+			};
+			var responseInfo2 = {
+				statusCode: 200,
+			};
+			var expectedResponse = {
+				statusCode: 200
+			};
+			config = config.extend({
+				request: {
+					formData: null
+				},
+				numMaxRetries: 1,
+				retryIntervalMS: 50,
+			});
+			requestStub.yieldsAsync(null, responseInfo1);
+			sandbox.stub(eventBusFake, 'emit').withArgs('response');
+
+			var apiRequest = new APIRequest(config, eventBusFake);
+			apiRequest.execute(function callback(err, response) {
+				assert.equal(apiRequest.numRetries, 1);
+				assert.equal(err, null);
+				assert.deepEqual(response, expectedResponse);
+				done();
+			});
+
+			setTimeout(function() {
+				requestStub.yieldsAsync(null, responseInfo2);
+			}, 30);
```

Comment:
Is it possible to set this mock in a different way without `setTimeout`? Something similar to `onCall` in `token-manager-test.js` to say that on the second call we return a different response?